### DISCUSSION TITLE
Schedule a task 3 days later.

### DISCUSSION THREAD

**Discussion by jymchng** [fastapi/fastapi#9347] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel

class Project(SQLModel):
    featured: bool = False
```

### Description

Let's say I have this sqlmodel.SQLModel which has a table name Project.

My intention is to feature a record of a Project for a definite period of time, e.g. 3 days, i.e. setting it's field featured to be True, and automatically set its field featured to be False thereafter.

How can I achieve this behavior using FastAPI?

### Operating System

Windows

### FastAPI Version

0.78.0

### Python Version

Python 3.10.6

**Comment by egisxxegis** [comment, marked as answer]
Good day, the FastAPI itself might not have methods to handle such case but You can mix FastAPI app with scheduler package.
For simple ones there are `apscheduler`, `rocketry`.

apscheduler:
https://stackoverflow.com/questions/75336685/how-to-create-scheduler-in-fastapi-uvicorn-server

rocketry:
https://rocketry.readthedocs.io/en/stable/cookbook/fastapi.html

Old thread:
https://github.com/tiangolo/fastapi/issues/520

fastapi-utils:
https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/

fastapi-scheduler:
https://pypi.org/project/fastapi-scheduler/

**Reply by jymchng** [reply]
Good day to you too sir, thank you for your prompt and extensive answer!
